cartography/intel/github/repos.py

```diff
@@ -178,6 +196,53 @@ def _transform_repo_languages(repo_url, repo, repo_languages):
             })
 
 
+def _transform_python_requirements(repo_object, out_requirements_files):
+    """
+    Performs data transformations for the requirements.txt and requirements3.txt files in a GitHub repo, if available.
+    :param repo_object: The repo object.
+    :param out_requirements_files: Output array to append transformed results to.
+    :return: Nothing.
+    """
+    for req_type in ['requirements', 'requirements3']:
+        req_file_contents = repo_object[req_type]
+        if req_file_contents and req_file_contents.get('text'):
+            # Remove --no-binary tags if present
+            text_contents = req_file_contents['text']
+            text_contents.replace('--no-binary ', '')
+
+            try:
+                parsed_list = requirements.parse(text_contents)
+            except ValueError as e:
+                logger.warning(
+                    f"Failed to parse {req_type}.txt in repo {repo_object['url']}, skipping."
+                    f"Details: {e}",
+                )
+                continue
+
+            for req in parsed_list:
+                if not req.name:
+                    logger.info(
+                        f"Could not get library name for an object in {repo_object['url']}'s {req_type}.txt, skipping.",
+                    )
+                    continue
+
+                spec = 'Unknown'
+                if len(req.specs) > 0:
+                    # We only want the version number not the specifier
+                    spec = req.specs[0][1]
+                elif req.revision:
+                    # For dynamic imports get the version from the uri if possible
+                    spec = req.revision
+
+                out_requirements_files.append({
```

Comment:
Not necessary in the first version, but it would be nice to include [extras](https://www.python.org/dev/peps/pep-0508/#extras) later on, which depending on the parser you may get for free.

Reply:
To be clear: I don't have a use case for extras right now. This will also likely complicate how we name graph nodes/where we put edges so probably best to save for a follow-up.